### PR TITLE
Integrate SGX SSL as a third party library

### PR DESCRIPTION
**Pull request by yang8621**
Signed-off-by: Yang Huang <yang.huang@intel.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -54,7 +63,31 @@ else ifeq ($(Build_Mode), SIM_PRERELEAESE)
 	@echo "The project has been built in simulation pre-release mode."
 endif
 
+ssl:
```
Comment (syan10) [inline Makefile:66]: can we move the this into the sub folder of SSL?

File: buildenv.mk [hunk 2]
```diff
@@ -69,6 +69,16 @@ CC ?= gcc
 CXX ?= g++
 RM = rm -f
 
+######## SGX SSL Settings ########
```
Comment (syan10) [inline buildenv.mk:72]: same here, it's specific for ssl

File: core/App/ehsm_provider.cpp [hunk 3]
```diff
@@ -958,7 +958,7 @@ ehsm_status_t VerifyQuote(ehsm_data_t *quote,
     // e.g. You can get latest QvE ISVSVN in QvE Identity JSON file from
     // https://api.trustedservices.intel.com/sgx/certification/v2/qve/identity
     // Make sure you are using trusted & latest QvE ISV SVN as threshold
-    sgx_isv_svn_t qve_isvsvn_threshold = 3;
+    sgx_isv_svn_t qve_isvsvn_threshold = 5;
```
Comment (syan10) [inline core/App/ehsm_provider.cpp:962]: why change it?

Comment (yang8621) [reply core/App/ehsm_provider.cpp:962]: verify quote failed with the error:
ERROR [App/ehsm_provider.cpp(1048) -> VerifyQuote]: Error in Verify QvE report and identity failed. 0xe051

**Review by syan10 — APPROVED** [review]

**Pull request merged**